src/EditorFeatures/CSharp/Completion/CompletionProviders/XmlDocCommentCompletion/XmlDocCommentCompletionProvider.cs

```diff
@@ -220,15 +220,39 @@ private string AttributeSelector(XmlElementSyntax element, string attribute)
             return null;
         }
 
-        private IEnumerable<CompletionItem> GetTagsForProperty(IPropertySymbol symbol, TextSpan filterSpan, DocumentationCommentTriviaSyntax trivia)
+        private IEnumerable<CompletionItem> GetTagsForProperty(IPropertySymbol symbol, TextSpan filterSpan, DocumentationCommentTriviaSyntax trivia, SyntaxToken token)
         {
             var items = new List<CompletionItem>();
 
-            var typeParameters = symbol.GetTypeArguments().Select(p => p.Name).ToSet();
+            if (symbol.IsIndexer)
+            {
+                var parameters = symbol.GetParameters().Select(p => p.Name).ToSet();
 
-            RemoveExistingTags(trivia, typeParameters, x => AttributeSelector(x, "typeparam"));
+                // User is trying to write a name, try to suggest only names.
+                if (token.Parent.IsKind(SyntaxKind.XmlNameAttribute) ||
+                    (token.Parent.IsKind(SyntaxKind.IdentifierName) && token.Parent.IsParentKind(SyntaxKind.XmlNameAttribute)))
```

Comment:
Yes, he should call the extracted helper I mention with the property's parameters for the enumerable of parameters and the property's ContainingType's type parameters as the type parameters...parameter.